Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,204 @@ extension UnifiedSymbolGraph {
         self.orphanRelationships = remainingOrphans
     }
 
+    /// If there were overload groups created by ``SymbolGraph/createOverloadGroupSymbols()``,
+    /// recalculate their ``SymbolGraph/Symbol/OverloadData`` and store it in
+    /// ``UnifiedSymbolGraph/Symbol/unifiedMixins``.
+    internal func combineOverloadGroups() {
+        // There are three main situations where a symbol's overload data would be different:
```

Comment:
One question: It's very possible that I don't see the bigger picture here but it seems like all of the situations that needs post processing happen because `SymbolGraph/createOverloadGroupSymbols()` runs on each individual symbol graph and doesn't have the full unified context. 

Assuming, ` combineOverloadGroups()` always runs, would it make sense to defer some of the work from `createOverloadGroupSymbols()` to only do it once? AFAICT it looks like `OverloadData` is recomputed for all overloaded symbols so I _think_ we could skip doing it in `createOverloadGroupSymbols()`.

Also, since you call out that iterating over `relationshipsByLanguage` could have a performance impact, would it make sense to store all the `.overloadOf` relationships to the side—that is, not in `SymbolGraph/relationships`—so that `combineOverloadGroups()` could operate on a much smaller list?

That would mean that both places (below) that iterate over all relationships but only care about `.overloadOf` relationships can work on that smaller list:
```swift
for relationship in self.relationshipsByLanguage.flatMap(\.value)
    where relationship.kind == SymbolGraph.Relationship.Kind.overloadOf
{
    ...
}
```

```swift
relationshipsByLanguage[selector] = relationships.filter({ relationship in
    ...
})
```

Then at the end we'd append the filtered down `.overloadOf` relationships to all the relationships and remove the side storage.

Does that make sense?

Reply:
I could see creating a `createOverloadGroupSymbols` that runs on unified graphs in addition to the one that runs on individual graphs, but i feel like that creates a lot of duplicate work. We could also do the side-storage suggestion for within the graph collector itself, so that it can collect the `overloadOf` relationships ahead of time since it's going to be iterating over all the symbol graphs' relationships anyway. That way the `combineOverloadGroups` work could just take those pre-collected relationships and not need to do the advance scanning like you said. I'll take a look at working that in.

Reply:
On the other hand, i think it's worth investigating that in a future PR, so that this one doesn't go back for more rounds of review now that it's gotten a few approvals.